eng/scripts/aotcompatibility/Check-AOT-Compatibility.ps1

```diff
@@ -0,0 +1,117 @@
+param([string]$ServiceDirectory, [string]$PackageName, [string]$ExpectedWarningsFilePath)
+
+### Creating a test app ###
+
+Write-Host "Creating a test app to publish."
+
+$expectedWarningsFullPath = Join-Path -Path "..\..\..\..\sdk\$ServiceDirectory\" -ChildPath $ExpectedWarningsFilePath
+
+$folderPath = "\TempAotCompatFiles"
+New-Item -ItemType Directory -Path "./$folderPath" | Out-Null
+Set-Location "./$folderPath"
+
+$csprojFile = "aotcompatibility.csproj"
+
+$csprojContent = @"
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net7.0</TargetFramework>
```

Comment:
This needs to be net8.0 as soon as it is supported in the repo and pipelines. These checks are not very useful when targeting net7.0 because only a fraction of the errors get reported.